@here/harp-map-controls/lib/CameraKeyTrackAnimation.ts

```diff
@@ -0,0 +1,289 @@
+/*
+ * Copyright (C) 2020 HERE Europe B.V.
+ * Licensed under Apache 2.0, see full license in LICENSE
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import { GeoCoordinates, GeoCoordLike } from "@here/harp-geoutils";
+import { LookAtParams, MapView, MapViewEventNames, MapViewUtils } from "@here/harp-mapview";
+import { LoggerManager } from "@here/harp-utils";
+import THREE = require("three");
+
+const logger = LoggerManager.instance.create("CameraKeyTrackAnimation");
+
+export interface ControlPointOptions extends Partial<LookAtParams> {
+    /**
+     * @override
+     */
+    target: GeoCoordinates | GeoCoordLike;
+    /**
+     * A timestamp for the [[ControlPoint]], describes when during the animation the
+     * [[ControlPoint]] should be reached, in seconds
+     */
+    timestamp: number;
+    /**
+     * @override
+     */
+    tilt?: number;
+    /**
+     * @override
+     */
+    heading?: number;
+    /**
+     * @override
+     */
+    distance?: number;
+    /**
+     * A name for the ControlPoint
+     */
+    name?: string;
+}
+/**
+ * A ControlPoint used to defined the Track of the [[CameraKeyTrackAnimation]]
+ */
+export class ControlPoint
```

Comment:
Why is this a class and why do we need this in addition to ControlPointOptions?
Or maybe the other way around, why do we need ControlPointOptions?

Reply:
Its a class to get sane defaults and the ControlPointOptions are there to not have to define everything at once. 
Basically new ControlPoint({target: <my-geo-coordinate>} gives you a valid ControlPoint